hazelcast/src/main/java/com/hazelcast/map/impl/operation/QueryOperation.java

```diff
@@ -115,53 +123,56 @@ public void run() throws Exception {
         checkPartitionStateChanges(partitionService, partitionStateVersion);
     }
 
-    private void checkPartitionStateChanges(InternalPartitionService partitionService, int partitionStateVersion) {
-        if (partitionStateVersion != partitionService.getPartitionStateVersion()) {
-            getLogger().info("Partition assignments changed while executing query: " + predicate);
+    public static int getMaxResultLimit(NodeEngine nodeEngine) {
```

Comment:
it would be nice to have some JavaDoc here. moreover - should this be really a responsibility of QueryOperation?